src/ui/Base/QueryBuilder.ts

```diff
@@ -530,4 +535,13 @@ export class QueryBuilder {
 
     return this.facetRequests;
   }
+
+  /**
+   * Set the userActions.tagViewsOfUser field of the query.
+   * This will modify the {@link QueryResult.isUserActionView} if the user saw/opened the result.
+   * @param user The ID of the user we'd like to check the document views.
+   */
+  public setUserOfUserActions(user: string) {
```

Comment:
Wondering if this is really necessary ? Versus just accessing the public `userActions` property ?

For "queryContext", it makes sense to add `setter` kind of function, since it does a tiny bit more logic to merge with any existing context value.

This property is simple enough that it should not be needed I think, does not seem possible to have more than one "userID" for a given "userActions" query parameter.

Also the name of this method is quite a mouthful, and we'd have to think of something "simpler" for the name, so one less thing to think about ;)

Reply:
👍 I agree, I was unsure what was the preferred way to modify a Query Builder. Will change that.